### ISSUE TITLE
Upgrade docToolchain components to current versions

### ISSUE THREAD

**Issue by mh182** [docToolchain/docToolchain#1102]
This is a tracking issue to show the current status of upgrading different docToolchain components:
- [x] #517 
- [x] #1105 
- [x] ~~#1103~~ #662 
- [x] #1071  
- [x] Upgrade to [OpenAPI Generator 6.5](https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-gradle-plugin)

A decision we have to make: how long do we want to support Java 8?
- [Gretty 4](https://github.com/gretty-gradle-plugin/gretty) doesn' support Java 8

**Comment by PacoVK** [comment]
Java 8 LTS is [EOL](https://endoflife.date/java) and is in security patch only mode. 
Does the community really need Java 8 support on a supporting tool like docToolchain? Basically you could run at least a dockerized environment to generate the docs that has minimum Java 11+ installed.

**Comment by PacoVK** [comment]
Upgrade OpenApi Generator is in #1104 
Currently maximum supported Gradle version is 7.0.2 see #1104

**Comment by mh182** [comment]
> Java 8 LTS is [EOL](https://endoflife.date/java) and is in security patch only mode.
> Does the community really need Java 8 support on a supporting tool like docToolchain? Basically you could run at least a dockerized environment to generate the docs that has minimum Java 11+ installed.

Some people, especially the ones behind air-gapped environments, are stuck with ancient environments. My approach would be to drop Java 8 in case it makes us problems when upgrading to never Java versions, or if some plugins can't be upgraded due to Java 8.

Question: are there plugins which can't be used due to Java 8?

Thanks to your research, we have a current state of how could proceed:
- Due to the Asciidoctor Gradle plugin v3.*, we are [stuck with Gradle 7.0.2](https://github.com/asciidoctor/asciidoctor-gradle-plugin/issues/626) 
- Java 17 was first supported with [Gradle 7.3](https://docs.gradle.org/current/userguide/compatibility.html), so we are stuck with Java 11

Asciidoctor Gradle Plugin v3 is in maintenance mode only. Asciidoctor Gradle Plugin v4, which supports Gradle 8 and thus Java 17, is in alpha state.

So we have the following options:
- we wait until Asciidoctor Gradle Plugin v4 is released, then upgrade Gradle 8 and Java 17
- we test the maturity of [Asciidoctor Gradle Plugin deveopment-4.x branch](https://github.com/asciidoctor/asciidoctor-gradle-plugin/commits/development-4.x) and if it is good enough, we upgrade to the alpha version. 

The decision if we drop Java 8 can be done when we proceed in our upgrade process.

@PacoVK : would it be a lot of effort to open a feature branch with Asciidoctor Plugin v4 where we could see how mature it is?

**Comment by PacoVK** [comment]
Alright makes sense with those ari-gapped environments!

**To Java 8:** 
I would say it kind of blocks us already with certain dependencies. An example would be that the strucutrizr dependency:
we currently use `com.structurizr:structurizr-dsl:1.29.1` but latest version is `com.structurizr:structurizr-dsl:1.30.1` 
since `1.30.0` the `dependencyUpdates`Task breaks to due missing Java 8 support [see](https://github.com/docToolchain/docToolchain/actions/runs/4671270401/jobs/8272037206#step:6:69) 
I didn't test it yet if that is a real issue, or just the  `dependencyUpdates`Task that fails because of something. 

**To the Asciidoctor Plugin v4:**
I think we should give it a try. I can check if i can create another PR, where we can test and elaborate on that 👍

**Comment by mh182** [comment]
> **To Java 8:**
> I would say it kind of blocks us already with certain dependencies. An example would be that the strucutrizr dependency:
> we currently use `com.structurizr:structurizr-dsl:1.29.1` but latest version is `com.structurizr:structurizr-dsl:1.30.1`
> since `1.30.0` the `dependencyUpdates`Task breaks to due missing Java 8 support [see](https://github.com/docToolchain/docToolchain/actions/runs/4671270401/jobs/8272037206#step:6:69)
> I didn't test it yet if that is a real issue, or just the `dependencyUpdates`Task that fails because of something.

Thanks for the information. This means we have so far two dependencies which are affected by Java 8:
- Structurizr
- Gretty

My gut feeling tells me we should drop Java 8 with the support for Java 17.

> To the Asciidoctor Plugin v4:
> I think we should give it a try. I can check if i can create another PR, where we can test and elaborate on that +1

Thanks a lot. Just assign the issue to yourself, so no one else starts working on the issue in parallel.

**Comment by PacoVK** [comment]
i cannot assign myself, as i have no permissions to do so

**Comment by mh182** [comment]
Replaced #1103 with #662 due to a block in the Gradle 8 upgrade. An upgrade to Gradle 7.5 is enough to support Java 17.
